module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -404,48 +425,83 @@ public function getConfig($function, $params = null)
      * Check item location against list of configured locations
      * where holds should be offered
      *
-     * @param string $locationName locationName from getHolding
+     * @param string $locationId locationId from getHolding
      *
      * @return bool
      */
-    protected function isHoldable($locationName)
+    protected function isHoldable($locationId)
     {
         return !in_array(
-            $locationName,
+            $locationId,
             (array)($this->config['Holds']['excludeHoldLocations'] ?? [])
         );
     }
 
     /**
-     * Gets the location name from the /locations endpoint and sets
-     * the display name to discoveryDisplayName,  name, or code
-     * based on whichever is available first in that order.
+     * Gets locations from the /locations endpoint and sets
+     * an array of location IDs to display names.
+     * Display names are set from discoveryDisplayName, or name
+     * if discoveryDisplayName is not available.
      *
-     * @param string $locationId ID of a location from the
-     * /holdings-storage/holdings endpoint
+     * @return array
+     */
+    protected function getLocations()
+    {
+        $cacheKey = 'locationMap';
+        $locationMap = $this->getCachedData($cacheKey);
+        //die(var_dump($locationMap));
+        if (null === $locationMap) {
+            $locationMap = [];
+            foreach ($this->getPagedResults(
+                'locations', '/locations'
+            ) as $location) {
+                if (!empty($location->discoveryDisplayName)) {
```

Comment:
You might be able to simplify this whole if..else to:

<pre>
$locationMap[$location->id] = !empty($location->discoveryDisplayName)
    ? $location->discoveryDisplayName : $location->name;
</pre>

(if the empty check is absolutely required -- i.e. if you're not sure if it's going to be null or an empty value like `''` or `0`)

...or better yet to:

<pre>
$locationMap[$location->id] = $location->discoveryDisplayName ?? $location->name;
</pre>

(if you're only actually worried about unset/null scenarios, rather than empty value scenarios).